sim/src/service/engineService.test.cpp

```diff
@@ -61,3 +119,153 @@ TEST_F(EngineServiceTest, GetVersion) {
     ASSERT_TRUE(s.ok());
     ASSERT_EQ(versionResp.commit_hash(), GIT_HASH);
 }
+
+TEST_F(EngineServiceTest, ApplyAndGetSimDef) {
+    auto testSim = test_simulation::TestSimulation();
+    const auto& simDef = testSim.simDef;
+    auto applyResp = applySim(testSim.simDef);
+
+    // For some reason, protobuf doesn't implement equality on its message
```

Comment:
What about [MessageDifferencer](https://developers.google.com/protocol-buffers/docs/reference/cpp/google.protobuf.util.message_differencer)?

Reply:
Yeah, that works, resolved in 9cf1e7f